src/threading.ts

```diff
@@ -247,11 +270,18 @@ export class BytecodeStackFrame implements IStackFrame {
 
     // Run until we get the signal to return to the thread loop.
     while (!this.returnToThreadLoop) {
-      var op = code.readUInt8(this.pc);
-      if (!RELEASE && logging.log_level === logging.VTRACE) {
-        vtrace(`  ${this.pc} ${annotateOpcode(op, this, code, this.pc)}`);
+      var op = method.getOp(this.pc, code);
```

Comment:
Function executions will not account for long-running loops as you rightly said.

Counting basic block executions is not possible currently, because the basic blocks (traces) are being created lazily after reaching a threshold. And the number of opcode executions is the threshold :)

It is possible to eagerly create basic blocks for all methods, but that would introduce a lot of overhead; most of the methods executed don't run long nor frequently enough.